rts/Sim/Units/Scripts/CobThread.cpp

```diff
@@ -435,8 +442,10 @@ bool CCobThread::Tick()
 
 				// return to caller
 				pc = LocalReturnAddr();
-				dataStackSize = std::min(dataStackSize, LocalStackFrame());
-				callStackSize -= 1;
+				while (dataStack.size() > LocalStackFrame()) {
+					dataStack.pop_back();
+				}
```

Comment:
This looks safe and idiomatic, though consider just
```suggestion
				if (const auto lsf = LocalStackFrame(); dataStack.size() > lsf) {
					dataStack.resize(lsf);
				}
```